api.lua

```diff
@@ -512,58 +513,79 @@ local function make_line_texture(line, lineno, pos, line_width, line_height, cwi
 	for word_i, word in ipairs(line) do
 		local chars = { }
 		local ch_offs = 0
-		word = string.gsub(word, "%^[12345678abcdefgh]", {
-			["^1"] = string.char(0x81),
-			["^2"] = string.char(0x82),
-			["^3"] = string.char(0x83),
-			["^4"] = string.char(0x84),
-			["^5"] = string.char(0x85),
-			["^6"] = string.char(0x86),
-			["^7"] = string.char(0x87),
-			["^8"] = string.char(0x88),
-			["^a"] = string.char(0x8a),
-			["^b"] = string.char(0x8b),
-			["^c"] = string.char(0x8c),
-			["^d"] = string.char(0x8d),
-			["^e"] = string.char(0x8e),
-			["^f"] = string.char(0x8f),
-			["^g"] = string.char(0x90),
-			["^h"] = string.char(0x91)
-		})
 		local word_l = #word
 		local i = 1
+		local escape = 0
 		while i <= word_l  do
-			local wide_c
-			if "&#x" == word:sub(i, i + 2) then
-				local j = i + 3
-				local collected = ""
-				while j <= word_l do
-					local c = word:sub(j, j)
-					if c == ";" then
-						wide_c = collected
-						break
-					elseif c < "0" then
-						break
-					elseif "f" < c then
-						break
-					elseif ("9" < c) and (c < "a") then
-						break
+			local wide_type, wide_c = string.match(word:sub(i), "^&#([xu])(%x+);")
+			local c = word:sub(i, i)
+			local c2 = word:sub(i+1, i+1)
+
+			if escape > 0 then escape = escape - 1 end
+			if c == "^" and escape == 0 and c2:find("[1-8a-h]") then
+				c = ({
+					["1"] = string.char(0x81),
+					["2"] = string.char(0x82),
+					["3"] = string.char(0x83),
+					["4"] = string.char(0x84),
+					["5"] = string.char(0x85),
+					["6"] = string.char(0x86),
+					["7"] = string.char(0x87),
+					["8"] = string.char(0x88),
+					["a"] = string.char(0x8a),
+					["b"] = string.char(0x8b),
+					["c"] = string.char(0x8c),
+					["d"] = string.char(0x8d),
+					["e"] = string.char(0x8e),
+					["f"] = string.char(0x8f),
+					["g"] = string.char(0x90),
+					["h"] = string.char(0x91)
+				})[c2]
```

Comment:
Could as well group these while changing it?
```suggestion
				c = string.char(tonumber(c2,18)+0x80))
```

Reply:
I mean for me at least it communicates intent better (which is shifting byte by octodecimal / base-18 value), makes it way shorter and saves constructing table for every character.